### PR TITLE
cmake,run-make-check: always enable WITH_GTEST_PARALLEL

### PR DESCRIPTION
**Pull request by tchaikov**
now that https://github.com/google/gtest-parallel/pull/63 has been
merged, we can now use gtest-parallel with py2 and also py3.

Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
see also #22931

**Comment by tchaikov** [comment]
retest this please.

File: run-make-check.sh [hunk 1]
```diff
@@ -99,14 +99,10 @@ function run() {
...
-        # gtest-parallel requires Python 2
-        CMAKE_PYTHON_OPTS="-DWITH_GTEST_PARALLEL=ON"
-    else
-        CMAKE_PYTHON_OPTS="-DWITH_PYTHON2=OFF -DWITH_PYTHON3=ON -DMGR_PYTHON_VERSION=3 -DWITH_GTEST_PARALLEL=OFF"
+    CMAKE_PYTHON_OPTS="-DWITH_GTEST_PARALLEL=ON"
+    if ! type python2 > /dev/null 2>&1 ; then
+        CMAKE_PYTHON_OPTS+=" -DWITH_PYTHON2=OFF -DWITH_PYTHON3=ON -DMGR_PYTHON_VERSION=3"
     fi
-
     CMAKE_BUILD_OPTS="-DWITH_FIO=ON -DWITH_SEASTAR=ON"
```
Comment (smithfarm) [inline run-make-check.sh]: There is no longer any dependency between gtest-parallel and Python, right? Would it make sense to move the `-DWITH_GTEST_PARALLEL=ON` into `CMAKE_BUILD_OPTS`?

**Comment by smithfarm** [comment]
OK, I'll test it.

**Pull request merged**